extensions/mssql/src/objectExplorerNodeProvider/connection.ts

```diff
@@ -6,226 +6,93 @@
-		// determine whether the host has either a ',' or ':' in it
-		this.setHostAndPort(',');
-		this.setHostAndPort(':');
-	}
-
-	// set port and host correctly after we've identified that a delimiter exists in the host name
-	private setHostAndPort(delimeter: string): void {
-		let originalHost = this._host;
-		let index = originalHost.indexOf(delimeter);
-		if (index > -1) {
-			this._host = originalHost.slice(0, index);
-			this._knoxPort = originalHost.slice(index + 1);
-		}
-	}
-
-	public get user(): string {
-		return this.connectionInfo.options[constants.userPropName];
-	}
-
-	public get password(): string {
-		return this.connectionInfo.options[constants.passwordPropName];
-	}
-
-	public get knoxport(): string {
-		if (!this._knoxPort) {
-			this.ensureHostAndPort();
+	constructor(connectionInfo: sqlops.connection.Connection | sqlops.IConnectionProfile) {
+		if (!this.isValid(connectionInfo)) {
+			throw new Error(localize('invalidConnectionInfo', 'Invalid ConnectionInfo is provided.'));
 		}
-		return this._knoxPort;
-	}
-
-	private static getKnoxPortOrDefault(connInfo: sqlops.ConnectionInfo): string {
-		let port = connInfo.options[constants.knoxPortPropName];
-		if (!port) {
-			port = constants.defaultKnoxPort;
-		}
-		return port;
-	}
-
-	public get connecttimeout(): number {
-		let timeoutSeconds: number = this.connectionInfo.options['connecttimeout'];
-		if (!timeoutSeconds) {
-			timeoutSeconds = constants.hadoopConnectionTimeoutSeconds;
+		if ('id' in connectionInfo) {
+			this._sqlClusterConnProfile = connectionInfo;
+			this._sqlClusterConnObj = this.toConnObj(this._sqlClusterConnProfile);
+		} else {
+			this._sqlClusterConnObj = connectionInfo;
+			this._sqlClusterConnProfile = this.toConnProfile(this._sqlClusterConnObj);
 		}
-		// connect timeout is in milliseconds
-		return timeoutSeconds * 1000;
+		this._host = this._sqlClusterConnObj.options[constants.hostPropName];
+		this._port = this._sqlClusterConnObj.options[constants.knoxPortPropName];
+		this._user = this._sqlClusterConnObj.options[constants.userPropName];
+		this._pass = this._sqlClusterConnObj.options[constants.passwordPropName];
 	}
 
-	public get sslverification(): string {
-		return this.connectionInfo.options['sslverification'];
-	}
+	public get sqlClusterConnObj(): sqlops.connection.Connection { return this._sqlClusterConnObj; }
+	public get sqlClusterConnProfile(): sqlops.IConnectionProfile { return this._sqlClusterConnProfile; }
+	public get host(): string { return this._host; }
+	public get port(): string { return this._port || constants.defaultKnoxPort; }
+	public get user(): string { return this._user; }
+	public get pass(): string { return this._pass; }
 
-	public get groupId(): string {
-		return this.connectionInfo.options[constants.groupIdName];
+	public isMatch(obj: SqlClusterConnection | sqlops.ConnectionInfo): boolean {
+		if (!obj) { return false; }
+		let options1 = 'options' in obj ? obj.options : obj._sqlClusterConnObj.options;
```

Comment:
Since SqlClusterConnection is a class, you can just check `instanceof`. Using this will give you type narrowing so will do correct verification of available properties.

Also you shouldn't need to use the `_sqlClusterConnection` private field, can use public accessor instead.